sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/LockRenewalOperationTest.java

```diff
@@ -0,0 +1,225 @@
+
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.messaging.servicebus.models.LockRenewalStatus;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import reactor.core.publisher.Mono;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayDeque;
+import java.util.Deque;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Function;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+import static org.mockito.Mockito.atMost;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * Represents a renewal session or message lock renewal operation that.
+ */
+class LockRenewalOperationTest {
+    private static final String A_LOCK_TOKEN = "a-lock-token";
+
+    private final ClientLogger logger = new ClientLogger(LockRenewalOperationTest.class);
+    private LockRenewalOperation operation;
+
+    @Mock
+    private Function<String, Mono<Instant>> renewalOperation;
+
+    @BeforeEach
+    void beforeEach() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @AfterEach
+    void afterEach() {
+        if (operation != null) {
+            operation.close();
+        }
+    }
+
+    @ValueSource(booleans = {true, false})
+    @ParameterizedTest
+    void constructor(boolean isSession) {
+        // Arrange
+        final Duration renewalPeriod = Duration.ofSeconds(4);
+        final Instant lockedUntil = Instant.now().plus(renewalPeriod);
+        final Duration maxDuration = Duration.ofSeconds(20);
+        when(renewalOperation.apply(A_LOCK_TOKEN))
+            .thenReturn(Mono.fromCallable(() -> Instant.now().plus(renewalPeriod)));
+
+        // Act
+        operation = new LockRenewalOperation(A_LOCK_TOKEN, maxDuration, isSession, renewalOperation, lockedUntil);
+
+        // Assert
+        if (isSession) {
+            assertEquals(A_LOCK_TOKEN, operation.getSessionId());
+            assertNull(operation.getLockToken());
+        } else {
+            assertEquals(A_LOCK_TOKEN, operation.getLockToken());
+            assertNull(operation.getSessionId());
+        }
+
+        assertEquals(lockedUntil, operation.getLockedUntil());
+        assertEquals(LockRenewalStatus.RUNNING, operation.getStatus());
+        assertNull(operation.getThrowable());
+    }
+
+    /**
+     * Verify that when an error occurs, it is displayed.
+     */
+    @Test
+    void errors() throws InterruptedException {
+        // Arrange
```

Comment:
Are we also taking care of situation when , user complete a message but renew operation still try to renew and keep getting error in logs. 
After settlement of the message by user, How does that lock is not renewing automatically in background and filling the logs ?